rfcs/groups/groups.mmd

```diff
@@ -0,0 +1,102 @@
+sequenceDiagram
+  participant M as Existing<br>member (M)
+  participant MA as Existing<br>member<br>agent (MA)
+  participant A as Alice (A)
+  participant AA as Alice's<br>agent (AA)
+  participant BA as Bob's<br>agent (BA)
+  participant B as Bob (B)
+
+  note over A, AA: 1. create new group (no members)
+  A ->> AA: gidA? GNEW gInfo<br>(gidA - conn alias of this group for A,<br>can be generated by the agent.<br>Does it share namespace with connections?)
+  AA ->> A: gidA OK<br>(or GOK?)
+
+  note over A, BA: 2. add Bob to group
+  A ->> AA: gidA GADD idAB<br>(idAB - conn alias A has for B)
+  
+  note over AA: initiate internal connection gidAB for B in group
+
+  AA ->> BA: via idAB: G_INV gidABinv gMeta gInfo<br>(gMeta contains number of members - TBC what else<br>should not contain date of creation or who created it)
+  BA ->> B: idBA GINV invID gMeta gInfo
+  B ->> BA: gidB? GJOIN invID<br>(could be also GREJECT invID?)
+  BA ->> B: gidB OK
+
+  BA ->> AA: establish internal connection gidBA (using gidABinv) for A in group<br>(intermal means that it is not visible to the clients. Separate namespace?)
+ 
+  AA -> A: gidA GCON idAB
+
+  note over M, B: below steps happen for each existing member M
+
+  AA ->> BA: via gidBA: GM_ADD lvlAM midABM|_<br>(midABM - how A identifies member M to B,<br>lvlAM - connection level, 0 - direct, 1 - via smbdy etc.)
+  
+  note over BA: initiate connection idBM for direct messages and<br> internal gidBM for group messages between B and M
```

Comment:
why does A write to B via `gidBA`, not `gidAB`? why is direct connection created implicitly w/t client's decision?

Reply:
I'm struggling to understand notations a bit - is "internal" connection not a new "physical duplex" connection, but another way to refer to existing "direct" connection? If so maybe rename "internal connection" -> "group connection", "internal" is very misleading here. Or my understanding is incorrect

Reply:
even if "internal" connection is only logical, still A should use her own `gidAB` to write to B, not `gidBA`, correct?

Reply:
"internal" - another duplex connection, but not addressable by agent client, and only used by the agent internally.

> why is direct connection created implicitly w/t client's decision?

Discussed, needs to be clarified in the doc - in short, to have it available for when it is needed, as it can be long to create (because of async comms).